server/py/framework/db/sqldb/db.py

```diff
@@ -5995,6 +5994,39 @@ def list_alerts(
             alerts.append(alert)
         return alerts
 
+    def list_and_delete_project_alerts(
+        self,
+        session,
+        project: str,
+    ) -> list[str]:
+        # We need to list ids and names here
+        # ids to reuse later for cleaning up the cache
+        # names to make notification deletion more efficient as notification table has index on name and parent_id
+        alerts = (
+            session.query(AlertConfig.id, AlertConfig.name)
+            .filter(AlertConfig.project == project)
+            .all()
+        )
+
+        # If there are no alerts, return an empty list
+        if not alerts:
+            return []
+
+        # delete related notifications
+        for alert_id, alert_name in alerts:
+            self._delete_alert_notifications(
+                session=session,
+                name=alert_name,
+                alert_id=alert_id,
+                project=project,
+                commit=False,
+            )
+            # delete the alert config
+            self._delete(session, AlertConfig, id=alert_id)
```

Comment:
we need to support batch deletion, this is very excessive

Reply:
perhaps use `_delete_multi_objects`

Reply:
@alonmr In this case it won't help, especially that it's better to request data by indexes instead of just project